src/fountain.rs

```diff
@@ -925,6 +934,21 @@ mod tests {
         assert!(!decoder.validate(&part));
     }
 
+    #[test]
+    fn test_empty_decoder_empty_part() {
+        let mut decoder = Decoder::default();
+        let part = Part {
+            sequence: 0,
```

Comment:
This is a tiny bit misleading, as only one of the empty fields here will be tested.

But we can leave this for a future cleanup.

Reply:
Added a couple more checks